helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -101,6 +104,54 @@ public boolean create(String path, T record, int options) {
     return result._retCode == RetCode.OK;
   }
 
+  public AccessResult retryUntilCreated(String path, Object record, int options, final ZkSerializer zkSerializer) {
+    AccessResult result = new AccessResult();
+    CreateMode mode = AccessOption.getMode(options);
+    if (mode == null) {
+      LOG.error("Invalid create mode. options: " + options);
+      result._retCode = RetCode.ERROR;
+      return result;
+    }
+
+    boolean retry;
+    do {
+      retry = false;
+      try {
+        _zkClient.create(path, record, ZooDefs.Ids.OPEN_ACL_UNSAFE, mode, zkSerializer);
```

Comment:
This is different from what we discussed.

What we agreed on was to have another HelixZkClient that uses a "dummy" serializer, correct? And we add the serialization logic in `create(String path, T record, int options, ZkSerializer zkSerializer)`?

Reply:
The latest change has covered the usage of a dummy serializer zkClient, please help take a look and there're details I believe needs to be adjusted.